package.json

```diff
@@ -19,38 +20,44 @@
     "build:story": "npm run build -w story",
     "prestory": "npm run build:tts",
     "story": "npm run start -w story",
-    "lint": "eslint --ignore-pattern dist --ext .ts,.tsx . \"packages/**/{src,__tests__}/**\"",
+    "lint": "eslint packages/**/{src,__tests__}/**/*.{ts,tsx}",
+    "lint:fix": "eslint --fix packages/**/{src,__tests__}/**/*.{ts,tsx}",
     "prettier": "prettier --no-error-on-unmatched-pattern --write *.ts packages/**/{src,__tests__}/**/*.{ts,tsx}"
   },
   "devDependencies": {
-    "@babel/cli": "^7.23.4",
-    "@babel/core": "^7.23.7",
     "@babel/preset-env": "^7.23.7",
     "@babel/preset-react": "^7.23.3",
     "@babel/preset-typescript": "^7.23.3",
+    "@eslint/js": "^9.5.0",
+    "@knighted/duel": "^2.0.0-rc.1",
+    "@knighted/dump": "^1.0.3",
+    "@testing-library/dom": "^10.3.0",
     "@testing-library/jest-dom": "^6.4.5",
-    "@testing-library/react": "^15.0.6",
+    "@testing-library/react": "^16.0.0",
     "@testing-library/user-event": "^14.5.2",
+    "@types/eslint__js": "^8.42.3",
     "@types/jest": "^29.5.12",
-    "@types/react": "^18.3.1",
+    "@types/react": "^18.3.3",
     "@types/react-dom": "^18.3.0",
-    "@typescript-eslint/eslint-plugin": "^7.8.0",
-    "@typescript-eslint/parser": "^7.8.0",
     "@vitejs/plugin-react": "^4.2.1",
-    "babel-dual-package": "^1.1.2",
     "babel-jest": "^29.7.0",
-    "babel-plugin-react-remove-properties": "^0.3.0",
-    "codecov": "^4.0.0-0",
     "eslint": "^8.57.0",
-    "eslint-plugin-jsx-a11y": "^6.8.0",
-    "eslint-plugin-react": "^7.34.1",
-    "eslint-plugin-react-hooks": "^4.6.2",
+    "eslint-plugin-jsx-a11y": "^6.9.0",
+    "eslint-plugin-react": "^7.34.3",
+    "eslint-plugin-react-hooks": "^5.1.0-rc.0",
+    "globals": "^15.6.0",
     "http-server": "^14.1.1",
     "jest": "^29.7.0",
     "jest-environment-jsdom": "^29.7.0",
     "prettier": "^3.2.5",
     "ts-jest-resolver": "^2.0.1",
-    "typescript": "^5.4.5",
-    "vite": "^5.2.8"
+    "typescript": "^5.5.2",
+    "typescript-eslint": "^8.0.0-alpha.39"
+  },
+  "overrides": {
+    "react": "^19.0.0-rc.0",
+    "react-dom": "^19.0.0-rc.0",
+    "tough-cookie": "^5.0.0-rc.3",
+    "whatwg-url": "^14.0.0"
```

Comment:
These are for `punycode` deprecation warnings while running tests. There is still a deprecation while running storybook.